### PR TITLE
Remove test.check from runtime deps by adding gen-helpers namespace

### PR DESCRIPTION
**Pull request by kennyjwilli**

### CODE REVIEW

**Comment by abhi18av** [comment]
Hi @kennyjwilli , I'm just curious here - does this provide other benefits with `spec` as opposed to plainly using `test.check` ?

**Comment by kennyjwilli** [comment]
@abhi18av No - it uses test.check internally. This just removes the need to have test.check on the classpath at runtime - you typically don't need to generate things in production.

**Comment by abhi18av** [comment]
Oh, I see. Thanks for the explanation :) 

So, I gather that basically through this PR we're mostly optimizing the Classpath deps.

**Comment by kennyjwilli** [comment]
That is correct.

**Comment by awkay** [comment]
Hey there...this is actually on my TODO list.

Did you test that this approach works for CLJS, and that advanced builds don't end up with cruft in them?

**Comment by kennyjwilli** [comment]
@awkay I did not. Sounds like a good thing to do. I don't have the time to do that at the moment though.

**Comment by wilkerlucio** [comment]
@kennyjwilli hello, sorry the immense delay to respond on this, I just tried it, but I saw a few problems, when I try to run the tests I get:

```ERROR in (query-ast-roundtrip) (generators.cljc:515)
Uncaught exception, not in assertion.
expected: nil
  actual: java.lang.AssertionError: Assert failed: Args to tuple must be generators
(every? generator? generators)```

Also, CLJS doesn't compile for the same reason, I don't have the time to look into it, do you know what is happening here?

File: src/edn_query_language/core.cljc [hunk 1]
```diff
@@ -31,7 +30,7 @@
 
      ::gen-ident-value
      (fn gen-ident-value [_]
-       (gen/frequency [[15 gen/simple-type-printable]
+       (gen/frequency [[15 (gen/simple-type-printable)]
```
Comment (wilkerlucio) [inline src/edn_query_language/core.cljc:33]: this shouldn't have the list around, since `gen/simple-type-printable` is a generator already.

Comment (kennyjwilli) [reply src/edn_query_language/core.cljc:33]: This is calling into `clojure.spec.gen.alpha`, not test.check's generator namespace. Everything in the spec.gen namespace is a function. 
```clojure
(require '[clojure.spec.gen.alpha :as gen])
=> nil
(gen/generate (gen/simple-type-printable))
=> -0.90850830078125
(gen/generate gen/simple-type-printable)
java.lang.NullPointerException: 
```

**Comment by kennyjwilli** [comment]
> @kennyjwilli hello, sorry the immense delay to respond on this, I just tried it, but I saw a few problems, when I try to run the tests I get:

I'd need the full stack trace to figure out what's happening. It sounds like a bad call to `gen/tuple`. How would I repro this?

**Comment by kennyjwilli** [comment]
I see that the current version in master no longer requires the `edn-query-language.gen` which allowed you to [remove](https://github.com/edn-query-language/eql/commit/d9691d2e0b8d7d11b57efdba80eb6c78a2b8cf97) the hard dep on test.check. This PR would allow you to require `edn-query-language.gen` in the core ns if desired. If that is no longer desired then this PR is not needed.

File: src/edn_query_language/core.cljc (hunk 1 above)
Comment (wilkerlucio) [reply src/edn_query_language/core.cljc]: ah, you right, thanks for the clarification

**Comment by wilkerlucio** [comment]
Yeah, I intended to keep it separated for now, the pattern around specs may be revisited in the future.

**Pull request closed without merging**